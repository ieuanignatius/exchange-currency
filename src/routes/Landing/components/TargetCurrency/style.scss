@import '../../../../styles/colors';

.target-currency-card {
  border: 1px solid $color-border;

  .target-currency-container {
    border-right: 1px solid $color-border;
    padding: 0 5px;

    p {
      font-weight: normal;
    }

    .currency-tag {
      font-weight: 600;
    }

    .currency-equality {
      font-weight: normal;
      font-size: 12px;
    }
  }
  
  .target-currency-btn-container {
    text-align: center;
    align-self: center;
  }

  +.target-currency-card {
    margin-top: 15px;
  }
}