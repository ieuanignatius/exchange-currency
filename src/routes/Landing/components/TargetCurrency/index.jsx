import React, { PureComponent } from 'react';
import { arrayOf, string, object, number, func } from 'prop-types';

import { currency } from "../../../../core/utils";

import './style.scss';

//thanks to (https://openexchangerates.org/api/currencies.json)
import currencyNameList from '../../currencyAbbreviationList';


class TargetCurrency extends PureComponent {
  static propTypes = {
    baseCurrency: string,
    targetCurrencyList: arrayOf(string),
    initialValue: number,
    rates: object,
    handleDeleteTargetCurrency: func
  };

  render(){
    const { targetCurrencyList, baseCurrency, initialValue, rates, handleDeleteTargetCurrency } = this.props;

    return targetCurrencyList.map((currencyCode, currencyIndex) => {
      const currencyName = currencyNameList[currencyCode];
      const currencyRate = initialValue * rates[currencyCode];

      return (
        <div className="container card target-currency-card" key={currencyIndex}>
          <div className="row">
            <div className="col-xs-9 col-sm-10 col-md-11 col-lg-11 target-currency-container">
              <p className="currency-code">{currencyCode}</p>
              <p className="currency-value">{currency(currencyRate.toString())}</p>
              <div className="clear-both"/>
              <p className="currency-tag">{currencyCode} - {currencyName}</p>
              <p className="currency-equality">1 {baseCurrency} = {currencyCode} {currency(rates[currencyCode].toString())}</p>
            </div>
            <div
              onClick={handleDeleteTargetCurrency.bind(this, currencyIndex)}
              className="col-xs-3 col-sm-2 col-md-1 col-lg-1 target-currency-btn-container"
            >
              (-)
            </div>
          </div>
        </div>
      );
    });
  }
}

export default TargetCurrency;

