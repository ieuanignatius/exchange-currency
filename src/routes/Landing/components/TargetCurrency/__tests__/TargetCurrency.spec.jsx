import React from 'react';
import TargetCurrency from '../';
import { mount } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const props = {
  baseCurrency: 'USD',
  targetCurrencyList: ['IDR'],
  initialValue: 10,
  rates: {
    IDR: 14491
  },
  handleDeleteTargetCurrency: jest.fn()
};

test('Should render TargetCurrency correctly', () => {
  const tree = mount(<TargetCurrency {...props}/>);
  
  expect(tree).toMatchSnapshot();
});
