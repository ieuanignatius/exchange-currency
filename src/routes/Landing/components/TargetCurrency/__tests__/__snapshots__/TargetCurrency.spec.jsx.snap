// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TargetCurrency correctly 1`] = `
<TargetCurrency
  baseCurrency="USD"
  handleDeleteTargetCurrency={[MockFunction]}
  initialValue={10}
  rates={
    Object {
      "IDR": 14491,
    }
  }
  targetCurrencyList={
    Array [
      "IDR",
    ]
  }
>
  <div
    className="container card target-currency-card"
    key="0"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-9 col-sm-10 col-md-11 col-lg-11 target-currency-container"
      >
        <p
          className="currency-code"
        >
          IDR
        </p>
        <p
          className="currency-value"
        >
          144,910
        </p>
        <div
          className="clear-both"
        />
        <p
          className="currency-tag"
        >
          IDR
           - 
          Indonesian Rupiah
        </p>
        <p
          className="currency-equality"
        >
          1 
          USD
           = 
          IDR
           
          14,491
        </p>
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-1 col-lg-1 target-currency-btn-container"
        onClick={[Function]}
      >
        (-)
      </div>
    </div>
  </div>
</TargetCurrency>
`;
